File: examples/example-kubernetes-client-openapi-informer-kotlin/src/test/kotlin/micronaut/client/configmap/ConfigMapControllerTest.kt
package micronaut.client.configmap

import io.micronaut.kubernetes.client.openapi.informer.SharedIndexInformer
import io.micronaut.kubernetes.client.openapi.informer.SharedIndexInformerFactory
import io.micronaut.kubernetes.client.openapi.informer.cache.Indexer
import io.micronaut.kubernetes.client.openapi.model.V1ConfigMap
import io.micronaut.kubernetes.client.openapi.model.V1ObjectMeta
import io.micronaut.kubernetes.client.openapi.resolver.NamespaceResolver
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNull
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`

class ConfigMapControllerTest {

    private fun configMap(name: String) = V1ConfigMap().metadata(V1ObjectMeta().name(name).namespace("default"))

    private fun controller(vararg configMaps: V1ConfigMap): ConfigMapController {
        val indexer = mock(Indexer::class.java) as Indexer<V1ConfigMap>
        `when`(indexer.list()).thenReturn(configMaps.toList())
        for (configMap in configMaps) {
            `when`(indexer.getByKey("default/" + configMap.metadata!!.name)).thenReturn(configMap)
        }
        val informer = mock(SharedIndexInformer::class.java) as SharedIndexInformer<V1ConfigMap>
        `when`(informer.indexer).thenReturn(indexer)
        val factory = mock(SharedIndexInformerFactory::class.java)
        `when`(factory.getExistingSharedIndexInformer(V1ConfigMap::class.java, "default")).thenReturn(informer)
        val namespaceResolver = mock(NamespaceResolver::class.java)
        `when`(namespaceResolver.resolveNamespace()).thenReturn("default")
        return ConfigMapController(factory, namespaceResolver)
    }

    @Test
    fun testAllReturnsTheCachedConfigMaps() {
        val controller = controller(configMap("first"), configMap("second"))

        assertEquals(listOf("first", "second"), controller.all().map { it.metadata!!.name })
    }

    @Test
    fun testConfigMapIsLookedUpByKey() {
        val controller = controller(configMap("first"))

        assertEquals("first", controller.configMap("first")!!.metadata!!.name)
        assertNull(controller.configMap("missing"))
    }
}

File: examples/example-kubernetes-client-openapi-informer-python/src/test/python/micronaut/client/configmap/ConfigMapControllerTest.py
from micronaut.kubernetes.client.openapi.model import V1ConfigMap, V1ObjectMeta
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from micronaut.client.configmap.ConfigMapController import ConfigMapController


class InMemoryIndexer:
    """Stands in for the informer cache, keyed by namespace/name."""

    def __init__(self, config_maps: list[V1ConfigMap]):
        self.config_maps = config_maps

    def list(self) -> list[V1ConfigMap]:
        return self.config_maps

    def getByKey(self, key: str) -> V1ConfigMap | None:
        for config_map in self.config_maps:
            metadata = config_map.getMetadata()
            if metadata.getNamespace() + "/" + metadata.getName() == key:
                return config_map
        return None


class InMemoryInformer:
    def __init__(self, indexer: InMemoryIndexer):
        self.indexer = indexer

    def getIndexer(self) -> InMemoryIndexer:
        return self.indexer


class InMemoryInformerFactory:
    def __init__(self, informer: InMemoryInformer):
        self.informer = informer

    def getExistingSharedIndexInformer(self, api_type, namespace: str) -> InMemoryInformer:
        return self.informer


class FixedNamespaceResolver:
    def resolveNamespace(self) -> str:
        return "default"


def config_map(name: str) -> V1ConfigMap:
    return V1ConfigMap().metadata(V1ObjectMeta().name(name).namespace("default"))


def controller(*config_maps: V1ConfigMap) -> ConfigMapController:
    factory = InMemoryInformerFactory(InMemoryInformer(InMemoryIndexer(list(config_maps))))
    return ConfigMapController(factory, FixedNamespaceResolver())


@MicronautTest
class ConfigMapControllerTest:

    @Test
    def test_all_returns_the_cached_config_maps(self):
        names = [cm.getMetadata().getName() for cm in controller(config_map("first"), config_map("second")).all()]
        assert names == ["first", "second"]

    @Test
    def test_config_map_is_looked_up_by_key(self):
        ctrl = controller(config_map("first"))
        assert ctrl.config_map("first").getMetadata().getName() == "first"
        assert ctrl.config_map("missing") is None
